React week3 new branch

elicarcar opened the pull request:
Netlfiy => https://quirky-poincare-334b90.netlify.com/

ArcoMul commented on homework-submission/react-week3-v2/src/HomeworkWeek3/HomeWorkWeek3.js:
```diff
@@ -0,0 +1,79 @@
+
+function HomeWorkWeek3() {
+  const API = "https://uinames.com/api/?ext";
+  const [data, setData] = useState([]);
```
`data` is a very generic name, can you find a better describing variable name?

ArcoMul commented on homework-submission/react-week3-v2/src/HomeworkWeek3/HomeWorkWeek3.js:
```diff
@@ -0,0 +1,79 @@
+    return function cleanup() {
+      return fetchNewUser();
+    };
+  }, []);
```
Cool that you used `useCallback`. This code implements almost the same as on line 13 to 26 though. Try to find a way that you can use either the `useEffect` in both cases or this `useCallback` implementation (to not repeat code unnecessary)

ArcoMul commented on homework-submission/react-week3-v2/src/HomeworkWeek3/HomeWorkWeek3.js:
```diff
@@ -0,0 +1,79 @@
+      }
+      return fetchWithSpace();
+    }
+  };
```
Ah and here is another time almost the same implementation, try to abstract code to a function or to combine the three

elicarcar commented:
Hello @ArcoMul thank you for your feedback! I implemented all of them, fixed the issue wuth the page jumping around, changed the states name (it's more descriptive now i guess) and instead of writing the code three times, i'm invoking it everytime i need (this is a better solution i guess it keeps the whole code DRY)

ArcoMul commented:
You can just push your changes in the branch of this pull request. No need to open a new pull request. That way I just losing track of what has been said already and what not :-)

Could you do so? Either merge your new branch in this old one, or look into git cherry picking.
